warehouse/query-core/src/test/java/datawave/query/iterator/logic/AndOrIteratorIT.java

```diff
@@ -0,0 +1,847 @@
+
+        OrIterator union = new OrIterator(unionIncludes, unionExcludes);
+
+        Set<NestedIterator<Key>> includes = new HashSet<>();
+        includes.add(IndexIteratorBridgeTest.createIndexIteratorBridge("FIELD_A", uidsAll, true));
+        includes.add(union);
+
+        Map<String,Integer> indexOnlyCounts = new HashMap<>();
+        indexOnlyCounts.put("FIELD_A", 1);
+        // indexOnlyCounts.put("FIELD_B", 2);
+
+        AndIterator itr = new AndIterator(includes);
+        assertThrows(IterationInterruptedException.class, () -> driveIterator(itr, uidsAll, indexOnlyCounts));
+    }
+
+    // nested union of negated terms which is interrupted
+    // A && !(B || C)
+    @Test
+    void testNegatedNestedUnionOfIndexOnlyFieldsIsInterrupted() {
+        Set<NestedIterator<Key>> unionIncludes = new HashSet<>();
+        unionIncludes.add(IndexIteratorBridgeTest.createInterruptibleIndexIteratorBridge("FIELD_B", uidsOdd, true, 4));
+        unionIncludes.add(IndexIteratorBridgeTest.createInterruptibleIndexIteratorBridge("FIELD_C", uidsOdd, true, 4));
+        OrIterator union = new OrIterator(unionIncludes);
+
+        OrIteratorIT.seekIterators(unionIncludes);
+
+        Set<NestedIterator<Key>> includes = new HashSet<>();
+        includes.add(IndexIteratorBridgeTest.createIndexIteratorBridge("FIELD_A", uidsAll, true));
+
+        Map<String,Integer> indexOnlyCounts = new HashMap<>();
+        indexOnlyCounts.put("FIELD_A", 5);
+
+        AndIterator itr = new AndIterator(includes, Collections.singleton(union));
+        assertThrows(IterationInterruptedException.class, () -> driveIterator(itr, uidsEven, indexOnlyCounts));
+    }
+
+    // === test by volume ===
+
+    /**
+     * For when you absolutely positively gotta make sure nothing slipped through the cracks
+     */
+    // A && (B || C)
+    @Test
+    void testByVolumeIntersectionWithNestedUnion() throws IOException {
+        int max = 100;
+        for (int i = 0; i < max; i++) {
+            SortedSet<String> uidsA = randomUids(50);
+            SortedSet<String> uidsB = randomUids(50);
+            SortedSet<String> uidsC = randomUids(50);
+
+            // build uids using De Morgans
+            SortedSet<String> leftUids = intersect(uidsA, uidsB);
+            SortedSet<String> rightUids = intersect(uidsA, uidsC);
+            final SortedSet<String> expectedUids = union(leftUids, rightUids);
+
+            Map<String,Integer> expectedFieldCounts = new HashMap<>();
+            expectedFieldCounts.put("FIELD_A", uidsA.size());
+            expectedFieldCounts.put("FIELD_B", leftUids.size());
+            expectedFieldCounts.put("FIELD_C", rightUids.size());
+
+            driveIntersectionWithSimpleNestedUnion(uidsA, uidsB, uidsC, expectedUids, expectedFieldCounts);
+        }
+    }
+
+    // this test is used to find edge cases
+    // A && (B || !C)
+    @Disabled
+    @Test
+    void testByVolumeIntersectionWithNestedUnionThatContainsANegation() throws IOException {
```

Comment:
This is going to be solved later

Reply:
if this case doesn't have a solution is this code ready for merge?